swift-sdk/AuthManager.swift

```diff
@@ -0,0 +1,84 @@
+//
+//  Created by Jay Kim on 9/3/20.
+//  Copyright © 2020 Iterable. All rights reserved.
+//
+
+import Foundation
+
+@objc public protocol IterableInternalAuthManagerProtocol {
+    func getAuthToken() -> String?
+    func requestNewAuthToken()
+    func logoutUser()
+}
+
+class AuthManager: IterableInternalAuthManagerProtocol {
+    init(onAuthTokenRequestedCallback: (() -> String?)?, localStorage: LocalStorageProtocol) {
+        ITBInfo()
+        
+        self.onAuthTokenRequestedCallback = onAuthTokenRequestedCallback
+        self.localStorage = localStorage
+        
+        retrieveAuthToken()
+    }
+    
+    deinit {
+        ITBInfo()
+    }
+    
+    // MARK: - IterableInternalAuthManagerProtocol
+    
+    func getAuthToken() -> String? {
+        return authToken
+    }
+    
+    func requestNewAuthToken() {
+        authToken = onAuthTokenRequestedCallback?()
+        
+        storeAuthToken()
+    }
+    
+    func logoutUser() {
+        authToken = nil
+        
+        storeAuthToken()
+    }
+    
+    // MARK: - Auth Manager Functions
+    
+    func storeAuthToken() {
+        localStorage.authToken = authToken
+    }
+    
+    func retrieveAuthToken() {
+        authToken = localStorage.authToken
+    }
+    
+    // MARK: - Private/Internal
+    
+    private var authToken: String?
+    
+    private var localStorage: LocalStorageProtocol
+    
+    private let onAuthTokenRequestedCallback: (() -> String?)?
+    
+    private static func decodeExpirationDateFromAuthToken(_ authToken: String) -> Int? {
```

Comment:
I like that this method appears very isolated and has no dependencies. 

Can you please add tests for this?